fe/fe-core/src/main/java/com/starrocks/lake/vacuum/AutovacuumDaemon.java

```diff
@@ -241,6 +244,12 @@ private void vacuumPartitionImpl(Database db, OlapTable table, PhysicalPartition
             vacuumRequest.graceTimestamp = Math.min(vacuumRequest.graceTimestamp,
                     Math.max(GlobalStateMgr.getCurrentState().getClusterSnapshotMgr()
                             .getSafeDeletionTimeMs() / MILLISECONDS_PER_SECOND, 1));
+            if (vacuumImmediatelyPartition(partition)) {
```

Comment:
I think it is better to just affect the original graceTimestamp but not `ClusterSnapshotMgr.getSafeDeletionTimeMs()`? Because we want to test the GC after snapshot, keep it is more useful.